sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/RemoveRedundantAggregatesSuite.scala

```diff
@@ -153,8 +150,14 @@ class RemoveRedundantAggregatesSuite extends PlanTest {
     comparePlans(optimized, expected)
   }
 
+  test("Remove redundant aggregate - upper has contains foldable expressions") {
+    val originalQuery = x.groupBy('a, 'b)('a, 'b).groupBy('a)('a, TrueLiteral).analyze
```

Comment:
hmm, how does this work? `groupBy('a)('a, TrueLiteral)` is not group only and not literal-only.

Reply:
`RemoveRedundantAggregates` already supported before this PR.

Reply:
https://github.com/apache/spark/pull/35795 to address this.